Source/WebCore/html/PDFDocument.cpp

```diff
@@ -181,32 +181,40 @@ void PDFDocument::finishedParsing()
         sendPDFArrayBuffer();
 }
 
-void PDFDocument::sendPDFArrayBuffer()
+void PDFDocument::postMessageToIframe(const String& name, JSC::JSObject* data)
 {
-    using namespace JSC;
-
-    auto* frame = m_iframe->contentFrame();
-    // FIXME: https://bugs.webkit.org/show_bug.cgi?id=236668 - Use postMessage
-    auto openFunction = frame->script().executeScriptIgnoringException("PDFJSContentScript.open"_s).getObject();
-
     auto globalObject = this->globalObject();
     auto& vm = globalObject->vm();
+    JSC::JSLockHolder lock(vm);
+
+    JSC::JSObject* message = constructEmptyObject(globalObject);
+    message->putDirect(vm, vm.propertyNames->message, JSC::jsNontrivialString(vm, name));
+    if (data)
+        message->putDirect(vm, JSC::Identifier::fromString(vm, "data"_s), data);
+
+    auto* contentWindow = m_iframe->contentFrame()->window();
+    auto* contentWindowGlobalObject = m_iframe->contentDocument()->globalObject();
+
+    WindowPostMessageOptions options;
+    if (data)
+        options = WindowPostMessageOptions { "*"_s, Vector { JSC::Strong<JSC::JSObject> { vm, data } } };
```

Comment:
Should probably use starAtom() instead of `"*"_s`.

Reply:
Why are we passing `data` via WindowPostMessageOptions's transferrable? Why isn't it enough to pass them as part of `message.data` ?

Also, why set the target to "*" when there is data but not when there isn't?

Reply:
> Why are we passing data via WindowPostMessageOptions's transferrable? Why isn't it enough to pass them as part of message.data ?

We're passing an ArrayBuffer, so this is better for memory.

> Also, why set the target to "*" when there is data but not when there isn't?

Switched to "/"_s